### ISSUE TITLE
Unable to send Outlook email

### ISSUE THREAD

**Issue by Jasonx365** [go-gitea/gitea#32305]
### Description

In the previous version, the configured Outlook emails were sent normally, but after updating to the latest version, they cannot be sent.
<img width="1305" alt="image" src="https://github.com/user-attachments/assets/83abaec8-e449-4882-9e13-91216d3d1283">

### Gitea Version

docker image 1.22.3-rootless

### Can you reproduce the bug on the Gitea demo site?

Yes

### Screenshots

<img width="833" alt="image" src="https://github.com/user-attachments/assets/152498db-71eb-4f13-9d7d-50983de2abee">

### How are you running Gitea?

docker image 1.22.3-rootless

### Database

PostgreSQL

**Comment by wxiaoguang** [comment]
The message is quite clear: outlook says: "basic auth is disabled", but you are still using it, so you can't succeed to send email by your "basic auth"

**Comment by wxiaoguang** [comment]
There are details on Microsoft's site: https://learn.microsoft.com/en-us/exchange/client-developer/legacy-protocols/how-to-authenticate-an-imap-pop-smtp-application-by-using-oauth

----

You can use the OAuth authentication service provided by Microsoft Entra (Microsoft Entra) to enable your application connect with IMAP, POP, or SMTP protocols to access Exchange Online in Office 365. To use OAuth with your application, you need to:

* [Register your application](https://learn.microsoft.com/en-us/exchange/client-developer/legacy-protocols/how-to-authenticate-an-imap-pop-smtp-application-by-using-oauth#register-your-application) with Microsoft Entra.
* [Get an access token](https://learn.microsoft.com/en-us/exchange/client-developer/legacy-protocols/how-to-authenticate-an-imap-pop-smtp-application-by-using-oauth#get-an-access-token) from a token server.
* [Authenticate connection requests](https://learn.microsoft.com/en-us/exchange/client-developer/legacy-protocols/how-to-authenticate-an-imap-pop-smtp-application-by-using-oauth#authenticate-connection-requests) with an access token.


Maybe it needs some people to try and implement it.

**Comment by venuses** [comment]
In this link, [a new method](https://support.microsoft.com/en-us/office/pop-imap-and-smtp-settings-for-outlook-com-d088b986-291d-42b8-9564-9c414e2aa040) is provided. Is it currently supported?
<img width="652" alt="image" src="https://github.com/user-attachments/assets/2532b6ff-dddd-4655-8378-0de0fe8b3a66">

**Comment by wxiaoguang** [comment]
No, there is no such support. There are only a few auth methods supported.

![image](https://github.com/user-attachments/assets/91c3da3e-f66d-4b50-8e09-888f98bb3bd0)

**Comment by venuses** [comment]
So now I can only change to another outbox😅

**Comment by venuses** [comment]
OK,I'll see how to solve it.

**Comment by changchaishi** [comment]
@venuses Maybe you can try to use an SMTP-relay server to accomplish this.
https://github.com/bokysan/docker-postfix?tab=readme-ov-file#xoauth2_client_id-xoauth2_secret-xoauth2_initial_access_token-xoauth2_initial_refresh_token-and-xoauth2_token_endpoint
